diff -Naur nvidia-vaapi-driver-68efa33131745f1b2d530c64f6692f3993f3d53c.old/src/vabackend.c nvidia-vaapi-driver-68efa33131745f1b2d530c64f6692f3993f3d53c/src/vabackend.c
--- nvidia-vaapi-driver-68efa33131745f1b2d530c64f6692f3993f3d53c.old/src/vabackend.c	2024-10-04 11:45:30.145497569 +0200
+++ nvidia-vaapi-driver-68efa33131745f1b2d530c64f6692f3993f3d53c/src/vabackend.c	2024-10-04 11:45:40.165577593 +0200
@@ -2186,11 +2186,11 @@
 VAStatus __vaDriverInit_1_0(VADriverContextP ctx) {
     LOG("Initialising NVIDIA VA-API Driver: %lX", ctx->display_type);
 
-    //drm_state can be passed in with any display type, including X11. But if it's X11, we don't
-    //want to use the fd as it'll likely be an Intel GPU, as NVIDIA doesn't support DRI3 at the moment
+    //drm_state can be passed in with any display type, including X11. But if it's X11 or wayland, we 
+    //don't want to use the fd as it'll likely be an Intel GPU, as NVIDIA doesn't support DRI3 at the
+    //moment
     bool isDrm = ctx->drm_state != NULL && ((struct drm_state*) ctx->drm_state)->fd > 0 &&
-                 (((ctx->display_type & VA_DISPLAY_MAJOR_MASK) == VA_DISPLAY_DRM) ||
-                  ((ctx->display_type & VA_DISPLAY_MAJOR_MASK) == VA_DISPLAY_WAYLAND));
+                 ((ctx->display_type & VA_DISPLAY_MAJOR_MASK) == VA_DISPLAY_DRM);   
 
     pthread_mutex_lock(&concurrency_mutex);
     LOG("Now have %d (%d max) instances", instances, max_instances);
